extensions/java-joor-dsl/runtime/src/main/java/org/apache/camel/quarkus/dsl/java/joor/runtime/JavaJoorDslRecorder.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.dsl.java.joor.runtime;
+
+import java.lang.reflect.Modifier;
+
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+import org.apache.camel.CamelContext;
+import org.apache.camel.CamelContextAware;
+import org.apache.camel.RoutesBuilder;
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.spi.ResourceAware;
+import org.apache.camel.support.ResourceHelper;
+import org.apache.camel.util.ObjectHelper;
+
+@Recorder
+public class JavaJoorDslRecorder {
+    public void registerRoutesBuilder(RuntimeValue<CamelContext> context, String className, String location) throws Exception {
+        Class<?> clazz = Class.forName(className);
+        boolean skip = clazz.isInterface() || Modifier.isAbstract(clazz.getModifiers())
+                || Modifier.isPrivate(clazz.getModifiers());
+        // must have a default no-arg constructor to be able to create an instance
+        boolean ctr = ObjectHelper.hasDefaultNoArgConstructor(clazz);
+        if (ctr && !skip) {
+            // create a new instance of the class
+            try {
+                Object obj = context.getValue().getInjector().newInstance(clazz);
+                if (obj instanceof RoutesBuilder) {
+                    // inject context and resource
+                    CamelContextAware.trySetCamelContext(obj, context.getValue());
+                    ResourceAware.trySetResource(obj, ResourceHelper.fromString(location, ""));
+                    context.getValue().addRoutes((RoutesBuilder) obj);
+                }
+            } catch (Exception e) {
+                throw new RuntimeCamelException("Cannot create instance of class: " + className, e);
+            }
+        }
+    }
```

Comment:
Would there be value in logging/warning about non matched items ?

Reply:
Done, thx for the feedback